filebeat/harvester/log.go

```diff
@@ -132,29 +88,45 @@ func (h *Harvester) Harvest() {
 			return
 		}
 
+		// Update offset if complete line has been processed
+		h.SetOffset(h.GetOffset() + int64(bytesRead))
+
+		event := h.createEvent()
+
 		if h.shouldExportLine(text) {
-			event := &input.FileEvent{
-				EventMetadata: h.Config.EventMetadata,
-				ReadTime:      ts,
-				Source:        h.Path,
-				InputType:     h.Config.InputType,
-				DocumentType:  h.Config.DocumentType,
-				Offset:        h.GetOffset(),
-				Bytes:         bytesRead,
-				Text:          &text,
-				Fileinfo:      &info,
-				JSONFields:    jsonFields,
-				JSONConfig:    h.Config.JSON,
-			}
 
-			h.SpoolerChan <- event // ship the new event downstream
+			event.ReadTime = ts
+			event.Bytes = bytesRead
+			event.Text = &text
+			event.JSONFields = jsonFields
 		}
 
-		// Set Offset
-		h.SetOffset(h.GetOffset() + int64(bytesRead)) // Update offset if complete line has been processed
+		// Always send event to update state, also if lines was skipped
+		h.sendEvent(event)
 	}
 }
 
+// createEvent creates and empty event.
+// By default the offset is set to 0, means no bytes read. This can be used to report the status
+// of a harvester
+func (h *Harvester) createEvent() *input.FileEvent {
+	return &input.FileEvent{
+		EventMetadata: h.Config.EventMetadata,
+		Source:        h.Path,
+		InputType:     h.Config.InputType,
+		DocumentType:  h.Config.DocumentType,
+		Offset:        h.GetOffset(),
+		Bytes:         0,
+		Fileinfo:      &h.fileInfo,
+		JSONConfig:    h.Config.JSON,
+	}
+}
+
+// sendEvent sends event to the spooler channel
+func (h *Harvester) sendEvent(event *input.FileEvent) {
+	h.SpoolerChan <- event // ship the new event downstream
```

Comment:
what if spooler is not reading from channel due to being blocked in libbeat output? We need to 'unblock' here?